framework/src/media/MediaRecorderImpl.cpp

```diff
@@ -273,41 +278,40 @@ recorder_result_t MediaRecorderImpl::stop()
 	medvdbg("MediaRecorderImpl::stop()\n");
 	RecorderWorker& mrw = RecorderWorker::getWorker();
 	if (!mrw.isAlive()) {
-		return RECORDER_ERROR;
+		return RECORDER_ERROR_NOT_ALIVE;
 	}
-	mrw.enQueue(&MediaRecorderImpl::stopRecorder, shared_from_this(), 0);
 
+	mrw.enQueue(&MediaRecorderImpl::stopRecorder, shared_from_this(), RECORDER_OK);
 	return RECORDER_OK;
 }
 
-void MediaRecorderImpl::stopRecorder(int errcode)
+void MediaRecorderImpl::stopRecorder(recorder_result_t ret)
 {
 	medvdbg("stopRecorder mCurState : %d\n", (recorder_state_t)mCurState);
 
-	if (errcode == 0 && mCurState != RECORDER_STATE_RECORDING && mCurState != RECORDER_STATE_PAUSED) {
-		notifyObserver(OBSERVER_COMMAND_ERROR);
+	if (ret == RECORDER_OK && mCurState != RECORDER_STATE_RECORDING && mCurState != RECORDER_STATE_PAUSED) {
```

Comment:
Please Add bracket for each state & please add description about this state, below is example.
/* Stop called normally but wrong state */